### ISSUE TITLE
linter: import-plugin lints unexpected files

### ISSUE THREAD

**Issue by spmiller** [oxc-project/oxc#8631]
### What version of Oxlint are you using?

latest

### What command did you run?

npx oxlint@latest --import-plugin -A all

### What does your `.oxlint.json` config file look like?

No config file

### What happened?

Enabling the `import-plugin` starts linting files in the `node_modules` directory which cannot be disabled. Errors are not detected when the linter is run without the `import-plugin` enabled.

Steps to reproduce:

1. Add a library with code that triggers a linting error (we used [temp@0.8.3](https://www.npmjs.com/package/temp))
2. Include at least one TS file that imports the library (the bug is not triggered without an import; both .ts and .mjs files trigger the issue)
3. Run `oxlint` enabling the `import-plugin`

Expected output:
```
Finished in 12ms on 1 file with 0 rules using 16 threads.
Found 0 warnings and 0 errors.
```

Actual output:
```
  × '0'-prefixed octal literals and octal escape sequences are deprecated
     ╭─[node_modules/temp/lib/temp.js:226:21]
 225 │   var dirPath = generateName(affixes, 'd-');
 226 │   fs.mkdir(dirPath, 0700, function(err) {
     ·                     ────
 227 │     if (!err) {
     ╰────
  help: for octal literals use the '0o' prefix instead

....

Finished in 11ms on 1 file with 0 rules using 16 threads.
Found 0 warnings and 5 errors.
```

Using `--ignore-pattern` does not seem to have any effect, e.g. the bug is still reproduced when using `--ignore-pattern "**/*.js"` or `--ignore-pattern "node_modules`.

`package.json`:
```
{
  "name": "repo",
  "version": "1.0.0",
  "dependencies": {
    "temp": "^0.8.3"
  }
}
```

`test.ts`:
```
import * as temp from 'temp';
```

**Comment by shulaoda** [comment]
I'm unable to reproduce your issue. It seems to have been fixed in the latest version. Could you try again, or provide more detailed information?

**Comment by Boshen** [comment]
There's a race condition in the multi-file runtime.

**Comment by camchenry** [comment]
Closing as completed in https://github.com/oxc-project/oxc/pull/9383

**Comment by spmiller** [comment]
Many thanks @Boshen for your hard work. I really appreciate your efforts.
